variation/regex.py

```diff
@@ -0,0 +1,215 @@
+
+from variation.schemas.app_schemas import AmbiguousRegexType
+from variation.schemas.classification_response_schema import (
+    AmbiguousType,
+    ClassificationType,
+)
+from variation.schemas.token_response_schema import TokenType
+
+CDNA_GENOMIC_SUBSTITUTION = re.compile(
+    r"^(?P<pos>\d+)(?P<ref>[ACTGN])>(?P<alt>[ACTGN])$"
+)
+
+CDNA_GENOMIC_REFERENCE_AGREE = re.compile(r"^(?P<pos>\d+)=$")
+
+CNDA_GENOMIC_DELETION = re.compile(
+    r"^(?P<pos0>\d+)(_(?P<pos1>\d+))?del(?P<deleted_sequence>[ACTGN]+)?$"
+)
+
+GENOMIC_DELETION_AMBIGUOUS_1 = re.compile(
+    r"^\((?P<pos0>\?|\d+)_(?P<pos1>\?|\d+)\)_\((?P<pos2>\?|\d+)_(?P<pos3>\?|\d+)\)del$"
+)
+
+GENOMIC_DELETION_AMBIGUOUS_2 = re.compile(
+    r"^\((?P<pos0>\?|\d+)_(?P<pos1>\?|\d+)\)_(?P<pos3>\d+)del$"
+)
+
+GENOMIC_DELETION_AMBIGUOUS_3 = re.compile(
+    r"^(?P<pos0>\d+)_\((?P<pos2>\?|\d+)_(?P<pos3>\?|\d+)\)del$"
+)
+
+CDNA_GENOMIC_DELINS = re.compile(
+    r"^(?P<pos0>\d+)(_(?P<pos1>\d+))?delins(?P<inserted_sequence>[ACTGN]+)$"
+)
+
+CDNA_GENOMIC_INSERTION = re.compile(
+    r"^(?P<pos0>\d+)_(?P<pos1>\d+)ins(?P<inserted_sequence>[ACTGN]+)$"
+)
+
+PROTEIN_SUBSTITUTION = re.compile(
+    r"^(?P<ref>[a-zA-z]+)(?P<pos>\d+)(?P<alt>([a-zA-Z]|Ter|\*)+)$"
+)
+
+PROTEIN_INSERTION = re.compile(
+    r"^(?P<aa0>[a-zA-z]+)(?P<pos0>\d+)_(?P<aa1>[a-zA-z]+)(?P<pos1>\d+)ins(?P<inserted_sequence>[a-zA-z]+)$"  # noqa: E501
+)
+
+PROTEIN_DELINS = re.compile(
+    r"^(?P<aa0>[a-zA-z]+)(?P<pos0>\d+)(_(?P<aa1>[a-zA-z]+)(?P<pos1>\d+))?delins(?P<inserted_sequence>[a-zA-z]+)$"  # noqa: E501
+)
+
+PROTEIN_DELETION = re.compile(
+    r"^(?P<aa0>[a-zA-z]+)(?P<pos0>\d+)(_(?P<aa1>[a-zA-z]+)(?P<pos1>\d+))?del(?P<deleted_sequence>[a-zA-z]+)?$"  # noqa: E501
+)
+
+PROTEIN_REFERENCE_AGREE = re.compile(r"^(?P<ref>[a-zA-z]+)(?P<pos>\d+)=$")
+
+GENOMIC_DUPLICATION = re.compile(r"^(?P<pos0>\d+)(_(?P<pos1>\d+))?dup$")
+
+# (#_#)_(#_#) OR (?_#)_(#_?)
+GENOMIC_DUPLICATION_AMBIGUOUS_1 = re.compile(
+    r"^\((?P<pos0>\?|\d+)_(?P<pos1>\?|\d+)\)_\((?P<pos2>\?|\d+)_(?P<pos3>\?|\d+)\)dup$"
+)
+
+# (?_#)_#, (#_?)_# OR (#_#)_#
+GENOMIC_DUPLICATION_AMBIGUOUS_2 = re.compile(
+    r"^\((?P<pos0>\?|\d+)_(?P<pos1>\?|\d+)\)_(?P<pos2>\d+)dup$"
+)
+
+# #_(#_?) OR #_(#_#)
+GENOMIC_DUPLICATION_AMBIGUOUS_3 = re.compile(
+    r"^(?P<pos0>\d+)_\((?P<pos2>\?|\d+)_(?P<pos3>\?|\d+)\)dup$"
+)
+
+# (#_#)_(#_#) OR (?_#)_(#_?)
+GENOMIC_DELETION_AMBIGUOUS_1 = re.compile(
+    r"^\((?P<pos0>\?|\d+)_(?P<pos1>\?|\d+)\)_\((?P<pos2>\?|\d+)_(?P<pos3>\?|\d+)\)del$"
+)
+
+# (?_#)_#, (#_?)_# OR (#_#)_#
+GENOMIC_DELETION_AMBIGUOUS_2 = re.compile(
+    r"^\((?P<pos0>\?|\d+)_(?P<pos1>\?|\d+)\)_(?P<pos2>\d+)del$"
+)
+
+# #_(#_?) OR #_(#_#)
+GENOMIC_DELETION_AMBIGUOUS_3 = re.compile(
+    r"^(?P<pos0>\d+)_\((?P<pos2>\?|\d+)_(?P<pos3>\?|\d+)\)del$"
+)
+
+# _REGEXPRS are used to help group the regex pattern and associated token type and
+# classification type
+
+# Note: Order matters for regexprs
+PROTEIN_REGEXPRS: List[Tuple[any, TokenType, ClassificationType]] = [
```

Comment:
```suggestion
PROTEIN_REGEXPRS: List[Tuple[Any, TokenType, ClassificationType]] = [
```

+ for the other annotations in this module

Reply:
+ the import obvi

Reply:
One other thing: right now, these regex List objects (everything from line 96 down) seem to only be used in 1 place. If you think they might be reused in more places in the future then keeping them here is good, but otherwise it might be cleaner to locate them in the place where they're being called